src/proposals.rs

```diff
@@ -161,15 +185,21 @@ async fn poll_approved_proposals(
                     }
 
                     if found_proposal {
-                        state.last_processed_proposal_id += missing_proposals;
+                        if pending_proposals {
+                            last_checked_pending_id += missing_proposals;
+                        } else {
+                            state.last_processed_proposal_id += missing_proposals;
+                        }
+
+                        continue;
                     } else {
                         info!(
                             "no new proposals. last processed proposal ID: {}",
                             state.last_processed_proposal_id
                         );
-                    }
 
-                    break;
```

Comment:
So previously here, we'd always break the loop and wait for the next polling call when any missing props were found?

Reply:
Yes we probably should have continued on found and broke on missing but we were always breaking.